ui/components/app/detected-token/detected-token.js

```diff
@@ -51,16 +53,52 @@ const DetectedToken = ({ setShowDetectedTokens }) => {
   const dispatch = useDispatch();
   const trackEvent = useContext(MetaMetricsContext);
 
-  const chainId = useSelector(getCurrentChainId);
   const detectedTokens = useSelector(getDetectedTokensInCurrentNetwork);
   const networkClientId = useSelector(getSelectedNetworkClientId);
-
-  const [tokensListDetected, setTokensListDetected] = useState(() =>
-    detectedTokens.reduce((tokenObj, token) => {
-      tokenObj[token.address] = { token, selected: true };
-      return tokenObj;
-    }, {}),
+  const detectedTokensMultichain = useSelector(
+    getAllDetectedTokensForSelectedAddress,
   );
+  const { tokenNetworkFilter } = useSelector(getPreferences);
+  const allNetworksFilterShown = Object.keys(tokenNetworkFilter ?? {}).length;
+
+  const configuration = useSelector(getNetworkConfigurationsByChainId);
+
+  const totalDetectedTokens = useMemo(() => {
+    return process.env.PORTFOLIO_VIEW && !allNetworksFilterShown
+      ? Object.values(detectedTokensMultichain).flat().length
+      : detectedTokens.length;
+  }, [detectedTokens, detectedTokensMultichain, allNetworksFilterShown]);
+
+  const [tokensListDetected, setTokensListDetected] = useState({});
+
+  useEffect(() => {
+    const newTokensList = () => {
+      if (process.env.PORTFOLIO_VIEW && !allNetworksFilterShown) {
+        return Object.entries(detectedTokensMultichain).reduce(
+          (acc, [chainId, tokens]) => {
+            if (Array.isArray(tokens)) {
+              tokens.forEach((token) => {
+                acc[token.address] = {
+                  token: { ...token, chainId },
+                  selected: tokensListDetected[token.address]?.selected ?? true,
+                };
+              });
+            }
+            return acc;
+          },
+          {},
+        );
+      }
+
+      return detectedTokens.reduce((tokenObj, token) => {
+        tokenObj[token.address] = { token, selected: true };
+        return tokenObj;
+      }, {});
+    };
+
+    setTokensListDetected(newTokensList());
+  }, [allNetworksFilterShown, detectedTokensMultichain, detectedTokens]);
```

Comment:
No need to make this change, but I wonder if it would be better to set `tokenListDetected` to the return of a `useMemo` function:
```
const tokenListDetected =   useMemo(() => {
    const newTokensList = () => {
      if (process.env.PORTFOLIO_VIEW && !allNetworksFilterShown) {
        return Object.entries(detectedTokensMultichain).reduce(
          (acc, [chainId, tokens]) => {
            if (Array.isArray(tokens)) {
              tokens.forEach((token) => {
                acc[token.address] = {
                  token: { ...token, chainId },
                  selected: tokensListDetected[token.address]?.selected ?? true,
                };
              });
            }
            return acc;
          },
          {},
        );
      }

      return detectedTokens.reduce((tokenObj, token) => {
        tokenObj[token.address] = { token, selected: true };
        return tokenObj;
      }, {});
    };


    return newTokensList()
  }, [allNetworksFilterShown, detectedTokensMultichain, detectedTokens]);
  ```
  ☝️ not tested